routes/auth_routes.js

```diff
@@ -0,0 +1,31 @@
+'use strict';
+
+const express = require('express');
+const bodyParser = require('body-parser').json();
+const User = require('../model/user');
+const basicHTTP = require('../lib/basic_http');
+
+const router = module.exports = exports = express.Router();
+
+router.post('/signup', bodyParser, (req, res, next) => {
+  let newUser = new User(req.body);
+  let hashedPassword = newUser.hashPassword();
+  newUser.password = hashedPassword;
+  req.body.password = null;
+  User.findOne({username: req.body.username}, (err, user) => {
+    if(err || user) return next(new Error('could not create user'));
+    newUser.save((err) => {
+      if (err) return next(new Error('could not create user'));
+      res.json({token: 'token'});
```

Comment:
This is where you should be generating a token.  The rest of this is spot on, you just need to be calling the function here that you set up in your schema.  Instead of responding with the string 'token'.